src/app/system/manage-data-tables/view-data-table/view-data-table.component.ts

```diff
@@ -0,0 +1,74 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatTableDataSource, MatPaginator, MatSort } from '@angular/material';
+
+/** Custom Services */
+import { SystemService } from '../../system.service';
+
+@Component({
+  selector: 'mifosx-view-data-table',
+  templateUrl: './view-data-table.component.html',
+  styleUrls: ['./view-data-table.component.scss']
+})
+
+/**
+ * View Data Table Component
+ */
+export class ViewDataTableComponent implements OnInit {
+
+  /** Data Table Data */
+  dataTableData: any;
+  /** Column Data */
+  columnsData: any;
+  /** Columns to be displayed in columns table. */
+  displayedColumns: string[] = ['columnName', 'columnType', 'columnLength', 'columnCode', 'isColumnPrimaryKey'];
+  /** Data source for columns table. */
+  dataSource: MatTableDataSource<any>;
+
+  /** Paginator for columns table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for columns table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the data table data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {SystemService} systemService System Service.
+   */
+  constructor(private route: ActivatedRoute,
+              private systemService: SystemService,
+              private router: Router) {
+    this.route.data.subscribe((data: { dataTable: any }) => {
+      this.dataTableData = data.dataTable;
+      this.columnsData = this.dataTableData.columnHeaderData;
+    });
+  }
+
+  /**
+   * Sets the columns table.
+   */
+  ngOnInit() {
+    this.setColumnsTable();
+  }
+
+  /**
+   * Initializes the data source, paginator and sorter for columns table.
+   */
+  setColumnsTable() {
+    this.columnsData.shift();
```

Comment:
Need of this line?

Reply:
One row of column data is not be required, therefore, deleting the same in this line.

Reply:
Why is that row coming? Could you elaborate a little.

Reply:
It is being generated in backend only (user is not passing it), hence coming from the backend (but not to be shown), for reference check it on the community-app.

Reply:
Handle this in the service.

Reply:
Add a similar `TODO` comment as I specified above.